### PR TITLE
import-external-cluster.sh: allow specifying the kubernetes context

### PR DESCRIPTION
**Pull request by raghavendra-talur**
If the import script is run on a system that has many kubernetes configs in the kubeconfig file, it is useful to be able to specify the context to use.

### CODE REVIEW

File: deploy/examples/import-external-cluster.sh [hunk 1]
```diff
@@ -65,11 +65,17 @@ function checkEnvVars() {
     echo "Providing both ROOK_EXTERNAL_ADMIN_SECRET and ROOK_EXTERNAL_USER_SECRET is not supported, choose one only."
     exit 1
   fi
+  if [ -n "$KUBECONTEXT" ]; then
```
Comment (parth-gr) [inline deploy/examples/import-external-cluster.sh:68]: Looks good,

Can you add some pointers of the changes in the documentation,
 https://github.com/rook/rook/blob/master/Documentation/CRDs/Cluster/external-cluster.md#commands-on-the-k8s-consumer-cluster
 
 When we would need this, and we need to set a extra env variable for this

**Review by parth-gr — CHANGES_REQUESTED** [review]

**Comment by parth-gr** [comment]
@raghavendra-talur any update, can you provide more information about the scenario when it is needed

**Comment by subhamkrai** [comment]
@raghavendra-talur any updates on this?

**Review by subhamkrai — APPROVED** [review]
lgtm

File: Documentation/CRDs/Cluster/external-cluster.md [hunk 2]
```diff
@@ -127,7 +127,9 @@ export RGW_POOL_PREFIX=default
 
 1. Paste the above output from `create-external-cluster-resources.py` into your current shell to allow importing the source data.
 
-2. Run the [import](https://github.com/rook/rook/blob/master/deploy/examples/import-external-cluster.sh) script.
+2. The import script in the next step uses the current kubeconfig context by default. If you want to specify the kubernetes cluster to use without changing the current context, you can specify the cluster name by setting the KUBECONTEXT environment variable.
```
Comment (parth-gr) [inline Documentation/CRDs/Cluster/external-cluster.md:130]: Maybe an example 
export KUBECONTEXT=?

File: deploy/examples/import-external-cluster.sh [hunk 3]
```diff
@@ -65,11 +65,17 @@ function checkEnvVars() {
...
+    echo "Using $KUBECONTEXT as the context value for kubectl commands"
+    KUBECTL="kubectl --context=$KUBECONTEXT"
```
Comment (parth-gr) [inline deploy/examples/import-external-cluster.sh:70]: I see we may do this by `config --kubeconfig=`, not sure what is the right way
https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/

**Review by parth-gr — CHANGES_REQUESTED** [review]
Some nis

File: deploy/examples/import-external-cluster.sh (hunk 3 above)
Comment (subhamkrai) [reply deploy/examples/import-external-cluster.sh:70]: generally `--context` is used with kubectl to apply the context for single use

File: Documentation/CRDs/Cluster/external-cluster.md (hunk 2 above)
Comment (raghavendra-talur) [reply Documentation/CRDs/Cluster/external-cluster.md]: Done

File: deploy/examples/import-external-cluster.sh (hunk 3 above)
Comment (raghavendra-talur) [reply deploy/examples/import-external-cluster.sh:70]: Yes, --context is the right one to use when there are multiple clusters within the same kubeconfig.

**Comment by raghavendra-talur** [comment]
@subhamkrai @parth-gr I believe the comments are addressed now.

**Comment by subhamkrai** [comment]
> @subhamkrai @parth-gr I believe the comments are addressed now.

Thanks @raghavendra-talur could you rebase once again, we fixed multiple ci issues past couple of days

**Comment by raghavendra-talur** [comment]
@subhamkrai Done.

**Review by subhamkrai — APPROVED** [review]

**Review by parth-gr — APPROVED** [review]
lgtm

**Pull request merged**